### ISSUE TITLE
[3.0.0-beta.2] fetch fails with POST requests

### ISSUE THREAD

**Issue by IAmPicard** [electron/electron#13618]
* Electron Version: 3.0.0-beta.2
* Operating System (Platform and Version): Windows 10
* Last known working Electron version: 2.0.3

**Expected Behavior**
window.fetch should work with POST

**Actual behavior**
TypeError: Failed to execute 'fetch' on 'Window': Request with GET/HEAD method cannot have body.

**To Reproduce**
Add this line of code:
```
// example uri for testing: 'https://httpbin.org/post'
let response = await fetch(uri, {method: 'POST', body: 'body'});
```
**Screenshots**
![image](https://user-images.githubusercontent.com/31076775/42546353-ed3c8fd4-8471-11e8-8bbd-47be31cf22db.png)

**Comment by GammaGames** [comment]
Mine is failing to send any body with jquery `$.post` as well

**Comment by computerquip-streamlabs** [comment]
I'm going to assume this is a chromium bug so https://github.com/electron/electron/issues/13683 might be interesting. Though, it's kinda crazy the fetch API is just broken in some cases. Looking at the chromium tree, looks like it's gone under serious changes in just the last few months as well.

**Comment by deepak1556** [comment]
@zcbenz the issue still happens with release build on windows.

**Comment by zcbenz** [comment]
This seems to be related to build environment. I can reproduce with official binaries, but not with my local binaries.

**Comment by zcbenz** [comment]
From `out/R/environment.x64`, I'm building with Windows SDK 10.0.17134.0. According to @deepak1556 the GN build also works fine, so I think Windows SDK version is very suspicious.


My build environment configuration:

<img width="328" alt="screen shot 2018-07-30 at 10 40 28" src="https://user-images.githubusercontent.com/639601/43373514-08306004-93e5-11e8-951e-32585c37867c.png">

**Comment by zcbenz** [comment]
This turns out to only happen when building for x64 target. I was building for ia32 target and it worked fine.

**Comment by brendtumi** [comment]
It's continue on 3.0.0-beta.3 too.

**Comment by yerol** [comment]
Fixed this temporarily by force-applying fetch polyfill. Removed the if check at the end of the polyfill for "self.fetch" and injected.

**Comment by IAmPicard** [comment]
Bug still repros with 3.0.0-beta.4

**Comment by ckerr** [comment]
Could this be the cause of https://github.com/electron/electron/issues/14004 as well?

**Comment by MarshallOfSound** [comment]
Fixed in #14022
